p2p/src/authenticated/relay.rs

```diff
@@ -0,0 +1,61 @@
+use futures::{channel::mpsc, SinkExt as _};
+
+#[derive(Clone)]
+pub struct Relay<T> {
```

Comment:
Curious for the rationale to make this generic? Assuming it is because you feel this sort of high/low prio channel may be useful elsewhere?

Reply:
I just thought it was cleaner to use a generic `T`